src/puppetlabs/puppetdb/jdbc.clj

```diff
@@ -41,7 +48,7 @@
   (sql/db-do-prepared *db* true (into [sql] params) {:multi? true}))
 
 (defn do-commands-outside-txn [& commands]
-  (let [conn (:connection *db*)
+  (let [conn ^Connection (:connection *db*)
```

Comment:
Seems a little odd to type-hint the right hand side of a let expression but it looks like it works.